docs/docs/topic-guides/custom-job-types.md

```diff
@@ -0,0 +1,90 @@
+and the requester node that receives the job submission.  
+
+:::info 
+As an experimental feature, the `exec` command is added to examine how the 
+feature can be used, extended and improved. It is possible that this command
+may be removed in future versions if an alternative approach is discovered to 
+be more effective.
+:::
+
+
+## CLI support 
+
+Within the CLI each custom job type requires a template file,
+a JSON representation of the job with a .tpl extension,
+found in the [templates folder](https://github.com/bacalhau-project/bacalhau/tree/main/cmd/cli/exec/templates). This template defines the base components of the job and is extended by the command line parameters provided to exec.  
+
+In addition to the usual runtime and specification flags, that can be found in [the CLI reference for exec](../references/cli/exec/), the `--code` parameter allows for single code files, or directories of code files to be added to the specification.  By default they will be added inline to the job specification, although the requester node may chose to change the storage provider for the code. There is however a hard-limit of 10MB for the attached code. 
+
+
+## Requester node
+
+When a job is received at the requester node, it determines how to process the job based
+on the engine type given in the job specification.  For engine types that are docker or 
+webassembly, they are processed immediately.  
+
+Other engine types are processed by the [translation package](https://github.com/bacalhau-project/bacalhau/blob/main/pkg/translation/translation.go) and tasks within the job translated by the server-side job translators, where one is provided for each custom job type.  These [translators](https://github.com/bacalhau-project/bacalhau/blob/main/pkg/translation/translation.go#L34C1-L38C2) are responsible for updating the task so that it is able to run on either the docker or webassembly runtimes.  
+
+Once a job has been translated, for example: Python -> Docker then the new job ID is returned to the client as if it were a normal job. The newly translated job may have a dependency on an external container which performs extra-steps during deployment, it is up to the translator to ensure it specifies the correct container or wasm module.
+
+
+## Initial job types 
+
+The initial implementation of custom job types includes job types for Python and DuckDB. Both of these custom job types support the `--code` parameter to simplify the code being sent to  the requester node - for python this is likely python code, for duckdb it is expected to be sql.  Command line parameters that follow the job type are used as the parameters to the translator, so `python /code/app.py /inputs/data.txt` will result in parameters that are `['python', '/code/app.py', '/inputs/data.txt']`.  Any flags that need to be passed to the python process, for instance should app.py require -f for the final parameter, must be sent after the bash -- separator.  
```

Comment:
Aside from reading these docs, how would a user find the list of custom job types bacalhau supports? Is there anything like a `list` command that returns these? Or maybe it's included in `exec --help`?

Reply:
Added a ticket to address this today